File: backend-java/sahayaSetu/src/main/java/com/sahayaSetu/controllers/AuthController.java
package com.sahayaSetu.controllers;

import com.sahayaSetu.dtos.*;
import com.sahayaSetu.services.IAuthService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/auth")
@RequiredArgsConstructor
public class AuthController {

    private final IAuthService authService;

    @PostMapping("/register")
    public ResponseEntity<?> registerDonor(@Valid @RequestBody DonorRegistrationDto dto) {
        return ResponseEntity.ok(authService.registerDonor(dto));
    }

    @PostMapping("/register-ngo")
    public ResponseEntity<?> registerNgo(@Valid @RequestBody NgoRegistrationDto dto) {
        return ResponseEntity.ok(authService.registerNgo(dto));
    }

    @PostMapping("/register-admin")
    public ResponseEntity<?> registerAdmin(@Valid @RequestBody AdminRegistrationDto dto) {
        return ResponseEntity.ok(authService.registerAdmin(dto));
    }

    @PostMapping("/login")
    public ResponseEntity<?> login(@Valid @RequestBody LoginDto dto) {
        return ResponseEntity.ok(authService.loginUser(dto));
    }

    @PostMapping("/login-ngo")
    public ResponseEntity<?> loginNgo(@Valid @RequestBody LoginDto dto) {
        return ResponseEntity.ok(authService.loginNgo(dto));
    }
}

File: backend-dotnet/SahayaSetu/Controllers/AuthController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using SahayaSetu.DTO.Auth;
using SahayaSetu.DTO.NGO;
using SahayaSetu.Services;

namespace SahayaSetu.Controllers
{
    [AllowAnonymous]
    [Route("api/[controller]")]
    [ApiController]
    public class AuthController : ControllerBase
    {
        private readonly IAuthService _authService;

        public AuthController(IAuthService authService)
        {
            _authService = authService;
        }

        [HttpPost("register")]
        public async Task<IActionResult> RegisterDonor([FromBody] DonorRegisterDto dto)
        {
            var result = await _authService.RegisterDonorAsync(dto);
            return Ok(result);
        }

        [HttpPost("register-admin")]
        public async Task<IActionResult> RegisterAdmin([FromBody] AdminRegisterDto dto)
        {
            var result = await _authService.RegisterAdminAsync(dto);
            return Ok(result);
        }

        [HttpPost("login")]
        public async Task<IActionResult> Login([FromBody] LoginDto dto)
        {
            var result = await _authService.LoginUserAsync(dto);
            return Ok(new 
            { 
                token = result.Token, 
                userId = result.Id,
                role = result.Role,
                name = result.Name,
                message = result.Message
            });
        }

        [HttpPost("register-ngo")]
        public async Task<IActionResult> RegisterNgo([FromBody] NgoRegisterDto dto)
        {
            var result = await _authService.RegisterNgoAsync(dto);
            return Ok(result);
        }

        [HttpPost("login-ngo")]
        public async Task<IActionResult> LoginNgo([FromBody] LoginDto dto)
        {
            var result = await _authService.LoginNgoAsync(dto);
            return Ok(new
            {
                token = result.Token,
                role = result.Role,
                name = result.Name,
                message = result.Message
            });
        }
    }
}
